src/Symfony/Component/HttpClient/RetryableHttpClient.php

```diff
@@ -28,24 +27,23 @@
  *
  * @author Jérémy Derussé <jeremy@derusse.com>
  */
-class RetryableHttpClient implements HttpClientInterface, ResetInterface
+class RetryableHttpClient implements HttpClientInterface, ResetInterface, LoggerAwareInterface
 {
     use AsyncDecoratorTrait;
+    use LoggerAwareTrait;
 
     private RetryStrategyInterface $strategy;
     private int $maxRetries;
-    private LoggerInterface $logger;
     private array $baseUris = [];
 
     /**
      * @param int $maxRetries The maximum number of times to retry
      */
-    public function __construct(HttpClientInterface $client, RetryStrategyInterface $strategy = null, int $maxRetries = 3, LoggerInterface $logger = null)
+    public function __construct(HttpClientInterface $client, RetryStrategyInterface $strategy = null, int $maxRetries = 3)
     {
         $this->client = $client;
         $this->strategy = $strategy ?? new GenericRetryStrategy();
         $this->maxRetries = $maxRetries;
-        $this->logger = $logger ?? new NullLogger();
```

Comment:
This change breaks existing code that passes a logger via the constructor and expects it to be used. We cannot do that.